google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -2887,4 +2953,77 @@ private void initMetricsCollection(MetricRegistry metricRegistry, List<LabelValu
         // TODO: Remove metric.
         ignored -> 0L);
   }
+
+  private void initOpenTelemetryMetricsCollection(
+      OpenTelemetry openTelemetry, Attributes attributes) {
+    if (openTelemetry == null) {
+      return;
+    }
+
+    Meter meter = openTelemetry.getMeter(MetricRegistryConstants.Scope);
+    meter
+        .gaugeBuilder(MAX_ALLOWED_SESSIONS_NAME)
+        .setDescription(MAX_ALLOWED_SESSIONS_DESCRIPTION)
+        .setUnit(COUNT)
+        .buildWithCallback(
+            measurement -> {
+              measurement.record(options.getMaxSessions(), attributes);
```

Comment:
nit: Can we add a small comment here noting that 'although this is a constant value, OT requires us to define this as a callback instead of setting a constant only once' (or something similar)

Reply:
Done.